ix backend utils does not contain a count_tokens module

The `PromptBuilder` class uses the `num_tokens_from_messages` method imported from `ix.utils.count_takens` at the top of the [file](https://github.com/kreneskyp/ix/blob/master/ix/agents/prompt_builder.py#L4). 

However, this file doesn't actually exist in the master branch. This change was made as part of a ['deprecations' commit](https://github.com/kreneskyp/ix/commit/88ff7c9242d93fb1ac03d7eb71ee5dd661ae564f) straight to the master branch - I would advise having separate PRs for each change for an easier audit trail of changes, particularly deprecations that move from using one method to another. Would also have allowed us to see a failing test in the pipeline.

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/graphene_django/settings.py", line 78, in import_from_string
    module = importlib.import_module(module_path)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/var/app/ix/schema/__init__.py", line 5, in <module>
    from ix.schema.mutations.chat import TaskFeedbackMutation, AuthorizeCommandMutation
  File "/var/app/ix/schema/mutations/chat.py", line 7, in <module>
    from ix.task_log.tasks.agent_runner import (
  File "/var/app/ix/task_log/tasks/__init__.py", line 1, in <module>
    from ix.task_log.tasks.agent_runner import *
  File "/var/app/ix/task_log/tasks/agent_runner.py", line 3, in <module>
    from ix.agents.process import AgentProcess
  File "/var/app/ix/agents/process.py", line 7, in <module>
    from ix.agents.prompt_builder import PromptBuilder
  File "/var/app/ix/agents/prompt_builder.py", line 4, in <module>
    from ix.utils.count_tokens import num_tokens_from_messages
ModuleNotFoundError: No module named 'ix.utils.count_tokens'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/app/./manage.py", line 22, in <module>
    main()
  File "/var/app/./manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 404, in run_from_argv
    parser = self.create_parser(argv[0], argv[1])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 367, in create_parser
    self.add_arguments(parser)
  File "/usr/local/lib/python3.11/site-packages/graphene_django/management/commands/graphql_schema.py", line 19, in add_arguments
    default=graphene_settings.SCHEMA,
            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/graphene_django/settings.py", line 125, in __getattr__
    val = perform_import(val, attr)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/graphene_django/settings.py", line 64, in perform_import
    return import_from_string(val, setting_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/graphene_django/settings.py", line 87, in import_from_string
    raise ImportError(msg)
ImportError: Could not import 'ix.schema.schema' for Graphene setting 'SCHEMA'. ModuleNotFoundError: No module named 'ix.utils.count_tokens'.
make: *** [graphene_to_graphql] Error 1
```

--- a/ix/utils/count_tokens.py
+++ b/ix/utils/count_tokens.py
@@ -1,0 +1,77 @@
+import tiktoken
+
+
+def num_tokens_from_messages(messages, model="gpt-3.5-turbo-0301"):
+    """Returns the number of tokens used by a list of messages."""
+    try:
+        encoding = tiktoken.encoding_for_model(model)
+    except KeyError:
+        print("Warning: model not found. Using cl100k_base encoding.")
+        encoding = tiktoken.get_encoding("cl100k_base")
+    if model == "gpt-3.5-turbo":
+        print(
+            "Warning: gpt-3.5-turbo may change over time. Returning num tokens assuming gpt-3.5-turbo-0301."
+        )
+        return num_tokens_from_messages(messages, model="gpt-3.5-turbo-0301")
+    elif model == "gpt-4":
+        print(
+            "Warning: gpt-4 may change over time. Returning num tokens assuming gpt-4-0314."
+        )
+        return num_tokens_from_messages(messages, model="gpt-4-0314")
+    elif model == "gpt-3.5-turbo-0301":
+        tokens_per_message = (
+            4  # every message follows <|start|>{role/name}\n{content}<|end|>\n
+        )
+        tokens_per_name = -1  # if there's a name, the role is omitted
+    elif model == "gpt-4-0314":
+        tokens_per_message = 3
+        tokens_per_name = 1
+    else:
+        raise NotImplementedError(
+            f"""num_tokens_from_messages() is not implemented for model {model}. See 
+            https://github.com/openai/openai-python/blob/main/chatml.md for information 
+            on how messages are converted to tokens."""
+        )
+    num_tokens = 0
+    for message in messages:
+        num_tokens += tokens_per_message
+        for key, value in message.items():
+            num_tokens += len(encoding.encode(value))
+            if key == "name":
+                num_tokens += tokens_per_name
+    num_tokens += 3  # every reply is primed with <|start|>assistant<|message|>
+    return num_tokens
+
+
+def count_tokens(text, model="gpt-3.5-turbo-0301"):
+    """Returns the number of tokens in a string."""
+    try:
+        encoding = tiktoken.encoding_for_model(model)
+    except KeyError:
+        print("Warning: model not found. Using cl100k_base encoding.")
+        encoding = tiktoken.get_encoding("cl100k_base")
+    if model == "gpt-3.5-turbo":
+        print(
+            "Warning: gpt-3.5-turbo may change over time. Returning num tokens assuming gpt-3.5-turbo-0301."
+        )
+        return count_tokens(text, model="gpt-3.5-turbo-0301")
+    elif model == "gpt-4":
+        print(
+            "Warning: gpt-4 may change over time. Returning num tokens assuming gpt-4-0314."
+        )
+        return count_tokens(text, model="gpt-4-0314")
+    elif model == "gpt-3.5-turbo-0301":
+        tokens_per_message = (
+            4  # every message follows <|start|>{role/name}\n{content}<|end|>\n
+        )
+        tokens_per_name = -1  # if there's a name, the role is omitted
+    elif model == "gpt-4-0314":
+        tokens_per_message = 3
+        tokens_per_name = 1
+    else:
+        raise NotImplementedError(
+            f"""num_tokens_from_messages() is not implemented for model {model}. See 
+            https://github.com/openai/openai-python/blob/main/chatml.md for information 
+            on how messages are converted to tokens."""
+        )
+    return len(encoding.encode(text))

